iml-gui/crate/src/components/command_modal.rs

```diff
@@ -577,44 +649,40 @@ fn extract_uri_id<T: EndpointName>(input: &str) -> Option<u32> {
     })
 }
 
-fn to_load_cmd(model: &Model, cmd_id: u32) -> bool {
-    // load it if the command is not found or is not complete
-    if let Some(cmd) = model.commands.get(&cmd_id) {
-        !cmd.complete
+fn perform_click(select: &mut Select, id: TypedId) -> bool {
+    if select.contains(id) {
+        !select.0.remove(&id)
     } else {
-        true
+        select.0.insert(id)
     }
 }
 
+fn to_load_cmd(model: &Model, cmd_id: u32) -> bool {
+    // load it if the command is not found or is not complete
+    model.commands.get(&cmd_id).map(|c| !c.complete).unwrap_or(true)
+}
+
 fn to_load_job(model: &Model, job_id: u32) -> bool {
-    if let Some(job) = model.jobs.get(&job_id) {
-        // job.state can be "tasked", "pending" or "complete"
-        // if a job is errored or cancelled, it is also complete
-        job.state != "complete"
-    } else {
-        true
-    }
+    // job.state can be "tasked", "pending" or "complete"
+    // if a job is errored or cancelled, it is also complete
+    model.jobs.get(&job_id).map(|j| j.state != "complete").unwrap_or(true)
 }
 
 fn to_load_step(model: &Model, step_id: u32) -> bool {
-    if let Some(step) = model.steps.get(&step_id) {
-        // step.state can be "success", "failed" or "incomplete"
-        step.state != "success" && step.state != "failed"
-    } else {
-        true
-    }
+    // step.state can be "success", "failed" or "incomplete"
+    model
+        .steps
+        .get(&step_id)
+        .map(|s| s.state != "success" && s.state != "failed")
```

Comment:
```suggestion
        .map(|s| s.state == "incomplete")
```

Instead of checking what the state is not, it's simpler to check what it is.

Reply:
I was not sure, that the list of "success", "failed" and "incomplete" is exhaustive and is not subject for a change.